File: Java-Core/src/main/java/presto/value/Document.java
package presto.value;

import java.util.HashMap;
import java.util.Map;

import presto.grammar.Identifier;
import presto.runtime.Context;
import presto.type.DocumentType;


public class Document extends BaseValue
{
	Map<Identifier,IValue> members = new HashMap<Identifier,IValue>();
	
	public Document() {
		super(DocumentType.instance());
	}
	
    @Override
    public IValue getMember(Context context, Identifier name)
    {
        IValue result = members.get(name);
        if(result== null)
        {
            result = new Document();
            members.put(name, result);
        }
        return result;
    }


    public void SetMember(Identifier name, IValue value)
    {
    	members.put(name, value);
    }
}

File: Java-CodeServer/src/main/resources/js/prompto/value/Document.js
var Value = require("./Value").Value;
var DocumentType = require("../type/DocumentType").DocumentType;

function Document(value) {
    Value.call(this, DocumentType.instance);
    this.mutable = true;
    this.members = {};
    return this;
}

Document.prototype = Object.create(Value.prototype);
Document.prototype.constructor = Document;

Document.prototype.getMember = function(context, name) {
    var result = this.members[name] || null;
    if(result==null) {
        result = new Document();
        this.members[name] = result;
    }
    return result;
};

Document.prototype.setMember = function(context, name, value) {
    this.members[name] = value;
};

exports.Document = Document;


